utils/utils.go

```diff
@@ -162,145 +179,134 @@ func DownloadFile(ctx context.Context, rawURL string, logger logging.Logger) (ou
+}
 
-	resp, err := httpClient.Do(req)
-	if err != nil {
-		return "", errw.Wrapf(err, "downloading file %s", rawURL)
+// convert GCP URLs to the form that has an accept-range header. Leave other URLs unchanged.
+// will mutate the input in the rewrite case.
+// bool return is true if the URL was rewritten.
+func rewriteGCPDownload(orig *url.URL) (*url.URL, bool) {
+	//nolint:lll
+	// try this on a shell to understand why this function is necessary. the first command has accept-range, the second doesn't.
+	// note that the second URL still seems to *accept* a range header, it just doesn't advertise that, which breaks our getter library.
+	// curl -I -X HEAD 'https://storage.googleapis.com/packages.viam.com/apps/viam-server/viam-server-v0.96.0-aarch64?generation=1759865152533030&alt=media'
+	// curl -I -X HEAD 'https://storage.googleapis.com/download/storage/v1/b/packages.viam.com/o/apps%2Fviam-server%2Fviam-server-v0.96.0-aarch64?generation=1759865152533030&alt=media'
+	// We end up with the second form because we consume MediaLink from https://github.com/googleapis/google-cloud-go/blob/125c39d61e201297004af8cd5f84b09e3dd511fe/storage/storage.go#L1571
+	// This function transforms MediaLink into the 'public URL' format.
+	if orig.Hostname() != "storage.googleapis.com" {
+		return orig, false
 	}
-	defer func() {
-		errRet = errors.Join(errRet, resp.Body.Close())
-	}()
-
-	if resp.StatusCode < 200 || resp.StatusCode >= 400 {
-		return "", errw.Errorf("got response '%s' while downloading %s", resp.Status, parsedURL)
-	}
-
-	//nolint:gosec
-	if err := os.MkdirAll(ViamDirs.Tmp, 0o755); err != nil {
-		return "", err
+	match := storagePathRegex.FindStringSubmatch(orig.Path)
+	if match == nil {
+		return orig, false
 	}
+	bucket := match[1]
+	path := match[2]
+	orig.Path = fmt.Sprintf("/%s/%s", bucket, path)
+	orig.RawPath = "" // this causes go to compute it where necessary.
+	return orig, true
+}
 
-	out, err := os.CreateTemp(ViamDirs.Tmp, "*")
-	if err != nil {
-		return "", err
+// CreatePartialPath makes a path under cachedir/part. These get cleaned up by CleanPartials.
+func CreatePartialPath(rawURL string) string {
+	var urlPath string
+	if parsed, err := url.Parse(rawURL); err != nil {
+		urlPath = "UNPARSED"
+	} else {
+		urlPath = parsed.Path
 	}
-	// `closed` suppresses double-close
-	closed := false
-	defer func() {
-		if !closed {
-			errRet = errors.Join(errRet, out.Close())
-		}
-		if runtime.GOOS != "windows" {
-			// todo(windows): doc why we don't do this / test adding it back
-			errRet = errors.Join(errRet, SyncFS(out.Name()))
-		}
-		if err := os.Remove(out.Name()); err != nil && !os.IsNotExist(err) {
-			errRet = errors.Join(errRet, err)
-		}
-	}()
 
-	workers := &sync.WaitGroup{}
-	writer, cancelFunc := downloadProgressSetup(ctx, out, outPath, resp.ContentLength, logger, workers)
-	defer workers.Wait()
-	defer cancelFunc()
+	return path.Join(ViamDirs.Partials, hashString(rawURL, 7), last(strings.Split(urlPath, "/"), "")+".part")
```

Comment:
I assume this is to deal with Windows' absurd path length limit. Out of curiosity, how did we end up with 7 as the magic number?